index.js

```diff
@@ -690,13 +690,23 @@ Lexer.prototype = {
   code: function() {
     var captures;
     if (captures = /^(!?=|-)[ \t]*([^\n]+)/.exec(this.input)) {
-      this.consume(captures[0].length);
       var flags = captures[1];
-      captures[1] = captures[2];
-      var tok = this.tok('code', captures[1]);
+      var code = captures[2];
+      var parsed, shortened = 0;
+      if (this.interpolated) {
+        try {
+          parsed = characterParser.parseUntil(code, ']');
```

Comment:
How does this handle code like:

```jade
- var index = Math.floor(Math.random() * 3);
p #[strong= ['a', 'b', 'c'][index]];
```

We could use `characterParser.parseMax(code)` and then check that the next character is `]`?